src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,82 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
+
+                          // If there is an update available, return a response with status - Available and image uri set in the
+                          // response
+                          auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
+                          auto nodeId = handler->GetSubjectDescriptor().subject;
+                          CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
+                          if (CHIP_NO_ERROR != err) {
+                              LogErrorOnFailure(err);
+
+                              // Handle busy error separately as we have a query image response status that maps to busy
+                              if (err == CHIP_ERROR_BUSY) {
+                                  response.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                                  if (!response.delayedActionTime.HasValue()) {
+                                      response.delayedActionTime.SetValue(kDelayedActionTimeSeconds);
+                                  }
+                              } else {
+                                  handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
+                                  handle.Release();
+                                  gOtaSender.ResetState();
+                                  return;
+                              }
+                          }
+                          auto targetNodeId
+                              = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+
+                          char uriBuffer[kMaxBDXURILen];
+                          MutableCharSpan uri(uriBuffer);
+                          err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
+                          if (CHIP_NO_ERROR != err) {
+                              LogErrorOnFailure(err);
+                              handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
+                              handle.Release();
+                              gOtaSender.ResetState();
+                              return;
+                          }
+
+                          response.imageURI.SetValue(uri);
+                          handler->AddResponse(cachedCommandPath, response);
+                          handle.Release();
+                      } else {
+                          response.status = static_cast<OTAQueryStatus>([data.status intValue]);
+                      }
+                      handler->AddResponse(cachedCommandPath, response);
+                      handle.Release();
 
-                          errorHandler:^(NSError *) {
-                              // Not much we can do here
-                          }];
-    };
+                      // Do not reset state if we have an update available or are returning busy (since we are currently in a BDX
+                      // session)
```

Comment:
I don't think so.  The Busy could have been from the delegate, so we never even got here...

But again, we should be handling the "delegate returned error" case way before this.  And we should not even be reaching this code for the "busy" case, since it should also be handled somewhere much earlier....

Reply:
We will get here. after i remove lines 669. let me fix the PR

Reply:
rearchitected the code. comment is resolved.